rfc/2022/2022-07-20 - Update Build Directory Structure of vets-website.md

```diff
@@ -0,0 +1,61 @@
+
+1. Bundles and assets from isolated application builds will be isolated from other builds.
+2. Isolated application builds can be deployed independently of *full builds* (non-isolated application builds).
+
+Updating the `build/generated` directory to follow this structure means that *full builds* will need to exclude the entry files of isolated applications, which ultimately excludes their bundles from being generated. The Webpack config will need to be updated to account for this, along with knowing when and where to store bundles and assets from isolated application builds.
+
+In the future, the subdirectories of isolated application bundles can be moved to `build/dist` to follow a more standard naming convention. 
+
+### Updating the content build for the new directory structure
+
+The `content-build` repository is responsible for creating the HTML files that are served on the site. By default, the build process in the `content-build` repository uses the `build/generated` path when injecting script tags that reference bundles and assets from `vets-website` deployments. This will need to be modified, since assets could be loaded from `build/generated` or one of the subdirectories for isolated application builds, e.g. `build/generated/vaos`. To better determine which paths in the deployment buckets to load bundles and assets from on each page, a generic script that handles this logic can be created. The build process in the `content-build` repository can then be updated to inject that generic script on each page. This essentially transfers control of determining which assets get loaded to `vets-website`, making it unlikely that production would break for applications migrating to the isolated structure.
+
+
+### Updating CI for the new directory structure
+
+Since there will no longer be a true *full build* once isolated application bundles are stored in their own subdirectory, we'll need to modify the process of building and testing global changes in CI. It won't be possible to build and test every application in one build anymore. One approach that would allow us to build and test multiple builds simultaneously involves Yarn Workspaces. With Yarn workspaces, each isolated application will have its own `package.json` file in the root of its directory, e.g. `src/applications/vaos/package.json`. The `package.json` files can be used to run the commands for each directory in `src/applications` that contains a `package.json`. This means we can use a command like `yarn workspaces foreach` at the root level to run commands like `build`, `test:unit`, `cy:run` (run Cypress tests), etc. The `workspaces foreach` command also has a `--parallel` option that can run the commands in parallel.
```

Comment:
> Instead of a full build of all apps, can we get away with just a build of non-isolated apps? That would be a smaller/faster build which would get even smaller/faster over time. Eventually it would disappear completely.

Yeah that's one of the big benefits of updating the directory structure. Isolated apps would be removed from the full build, and the _full build_ essentially becomes the non-isolated apps build, which like you said, would continue shrinking over time. Although this means we'll have more isolated builds that need to run in parallel when platform or global changes are made

> Can Yarn Workspaces help us determine which files actually need to be built/deployed based on which files are modified? Or is that something only nx can do?

That's a good question 🤔 I'm not sure if Yarn Workspaces can make an advanced dependency graph like Nx can, but if the directories in `src/platform` are going to be turned into workspaces, we could potentially use that to our advantage in determining which apps need to be built/deployed based on which platform files are updated